### PR TITLE
[SPARK-35521][INFRA] List Python 3.8 installed libraries in build_and_test workflow

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?

In the build_and_test workflow, tests are run against both Python 3.6 and Python 3.8. However, only libraries installed in Python 3.6 are listed. We should list Python 3.8's installed libraries as well.

### Why are the changes needed?

Listing Python 3.8's installed libraries is helpful to debugging.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual check.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
CC @HyukjinKwon

**Comment by ueshin** [comment]
There is another PR to upgrade Python to 3.9 in which the library list will be shown. #32657

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -217,6 +217,11 @@ jobs:
       run: |
         python3.6 -m pip install numpy 'pyarrow<3.0.0' pandas scipy xmlrunner plotly>=4.8
         python3.6 -m pip list
+    - name: List Python packages (Python 3.8)
```
Comment (xinrong-databricks) [inline .github/workflows/build_and_test.yml:220]: The run takes 5s.

**Comment by HyukjinKwon** [comment]
Let's drop this for now since now we upgraded it to Python 3.9, and don't run the tests with Python 3.8 now.

**Pull request closed without merging**